client/src/schema/repository.ts

```diff
@@ -517,12 +539,25 @@ export class BaseClient<D extends Record<string, Repository<any>> = Record<strin
 
     this.db = new Proxy({} as D, {
       get: (_target, prop) => {
-        if (typeof prop !== 'string') throw new Error('Invalid table name');
+        if (!isString(prop)) throw new Error('Invalid table name');
         return factory.createRepository(this, prop);
       }
     });
   }
 
+  async search<Tables extends keyof D>(
+    query: string,
+    tables: Tables[] = Object.keys(this.db) as Tables[],
+    options?: { fuzziness?: number }
+  ): Promise<{ [Model in GetArrayInnerType<typeof tables>]: Awaited<ReturnType<D[Model]['search']>> }> {
+    // TODO: Implement global search with a single call, REST repository abstraction needed
```

Comment:
We need to revisit this, but a discussion on plugins/namespaces should happen before. We can discuss it in our next 1:1 or a team sync.

Reply:
I would like to offer both (inside and outside) as-is here.

Creating https://github.com/xataio/client-ts/issues/182 for improving this